flexget/components/notify/notifiers/gotify.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import base64
+import datetime
+import re
+
+from flexget import plugin
+from flexget.event import event
+from flexget.config_schema import one_or_more
+from flexget.plugin import PluginWarning
+from flexget.utils.requests import Session as RequestSession, TimedLimiter
+from requests.exceptions import RequestException
+
+plugin_name = 'gotify'
+log = logging.getLogger(plugin_name)
+
+requests = RequestSession(max_retries=3)
+
+gotify_url_pattern = {
+    'type': 'string',
+    'pattern': r'^http(s?)\:\/\/.*\/message$',
+    'error_pattern': 'Gotify URL must begin with http(s) and end with `/message`',
+}
+
+class GotifyNotifier(object):
+    """
+    Example::
+
+    notify:
+      entries:
+        via:
+          - gotify:
+              url: <GOTIFY_SERVER_URL>
+              token: <GOTIFY_TOKEN>
+              priority: <PRIORITY>
+    Configuration parameters are also supported from entries (eg. through set).
+    """
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': gotify_url_pattern,
+            'token': {'type': 'string'},
+            'priority': {'type': 'integer', 'default': 4},  
+        },  
+        'required': [
+            'token',
+            'url',
+        ],
+        'additionalProperties': False,
+    }
+
+    def notify(self, title, message, config):
+        """
+        Send a Gotify notification
+        """
+        url = config['url']
+        params = {'token': config['token']}
+        
+        if config['priority']:
+          priority = config['priority']
+        
+        requests = RequestSession(max_retries=3)
```

Comment:
you already defined this in the global scope, this is redundant

Reply:
https://github.com/Flexget/Flexget/pull/2533/commits/f658a4667bcc37a8b0dc456d77e3b3c67c50b843